zenodo/modules/utils/cli.py

```diff
@@ -26,28 +26,181 @@
 
 from __future__ import absolute_import, print_function
 
+import os
+from io import SEEK_END, SEEK_SET
+
 import click
 from flask_cli import with_appcontext
-from invenio_pidstore.resolver import Resolver
-from invenio_records.api import Record
+from invenio_db import db
+from invenio_files_rest.models import ObjectVersion
 
 from zenodo.modules.deposit.tasks import datacite_register
+from zenodo.modules.records.api import record_resolver
 
 
 @click.group()
 def utils():
     """Zenodo helper CLI."""
 
+
 @utils.command('datacite_register')
 @click.argument('recid', type=str)
 @click.option('--eager', '-e', is_flag=True)
 @with_appcontext
 def datecite_register(recid, eager):
     """Send a record to DataCite for registration."""
-    pid, record = Resolver(
-        pid_type='recid', object_type='rec',
-        getter=Record.get_record).resolve(recid)
+    pid, record = record_resolver.resolve(recid)
     if eager:
         datacite_register.s(pid.pid_value, str(record.id)).apply(throw=True)
     else:
         datacite_register.s(pid.pid_value, str(record.id)).apply_async()
+
+
+@utils.command('add_file')
+@click.argument('recid', type=str)
+@click.argument('fp', type=click.File('rb'))
+@with_appcontext
+def add_file(recid, fp):
+    """Add a new file to a publishd record."""
+    pid, record = record_resolver.resolve(recid)
+    bucket = record.files.bucket
+    key = os.path.basename(fp.name)
+
+    obj = ObjectVersion.get(bucket, key)
+    if obj is not None:
+        click.echo('File "{key}" alreay exists in the record {recid}'.format(
+            key=key, recid=recid))
+        return
```

Comment:
The primary use case is to update an existing file. So perhaps add cli option ``-f/--force`` which will overwrite an existing file.